independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/buildextension/beans/SyntheticInjectionPointListAllTest.java

```diff
@@ -0,0 +1,88 @@
+package io.quarkus.arc.test.buildextension.beans;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.List;
+
+import jakarta.enterprise.util.TypeLiteral;
+import jakarta.inject.Singleton;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.ClassType;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.ParameterizedType;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.arc.All;
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.BeanCreator;
+import io.quarkus.arc.SyntheticCreationalContext;
+import io.quarkus.arc.processor.BeanRegistrar;
+import io.quarkus.arc.test.ArcTestContainer;
+
+public class SyntheticInjectionPointListAllTest {
+
+    @RegisterExtension
+    public ArcTestContainer container = ArcTestContainer.builder()
+            .beanClasses(SomeBean.class)
+            .beanRegistrars(new TestRegistrar()).build();
+
+    @Test
+    public void testListAllInjection() {
+        SyntheticBean synthBean = Arc.container().instance(SyntheticBean.class).get();
+        assertNotNull(synthBean);
+        List<SomeBean> list = synthBean.getList();
+        assertNotNull(list);
+        assertTrue(list.size() == 1);
+    }
+
+    @Singleton
+    static class SomeBean {
+
+        public String ping() {
+            return SomeBean.class.getSimpleName();
+        }
+
+    }
+
+    static class SyntheticBean {
+
+        private List<SomeBean> list;
+
+        public SyntheticBean(List<SomeBean> list) {
+            this.list = list;
+        }
+
+        public List<SomeBean> getList() {
+            return list;
+        }
+    }
+
+    static class TestRegistrar implements BeanRegistrar {
+
+        @Override
+        public void register(RegistrationContext context) {
+            context.configure(SyntheticBean.class)
+                    .addType(ClassType.create(DotName.createSimple(SyntheticBean.class)))
+                    .creator(SynthBeanCreator.class)
+                    // add injection point for @All List<SomeBean>
+                    .addInjectionPoint(ParameterizedType.create(List.class, ClassType.create(SomeBean.class)),
+                            AnnotationInstance.builder(All.class).build())
+                    .unremovable()
+                    .done();
+        }
+
+    }
+
+    public static class SynthBeanCreator implements BeanCreator<SyntheticBean> {
+
+        @Override
+        public SyntheticBean create(SyntheticCreationalContext<SyntheticBean> context) {
+            return new SyntheticBean(context.getInjectedReference(new TypeLiteral<>() {
```

Comment:
Hm, this is translated to the required type `Object`, or? So I wonder how the test could pass :thinking:.

Reply:
The `SyntheticBean` constructor takes a `List<SomeBean>`, so the type argument here is inferred to be `List<SomeBean>`.

Reply:
Ah! Not sure if inferring makes the code more readable here though.

Reply:
> Ah! Not sure if inferring makes the code more readable here though.

I can see why it is confusing - I had the type here explicitly but IDEA told me it isn't needed.
Let me add it back :)